matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+import com.powsybl.iidm.ConversionParameters;
+import com.powsybl.iidm.import_.Importer;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.parameters.Parameter;
+import com.powsybl.iidm.parameters.ParameterDefaultValueConfig;
+import com.powsybl.iidm.parameters.ParameterType;
+import com.powsybl.matpower.model.*;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.ConnectivityInspector;
+import org.jgrapht.graph.Pseudograph;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.Month;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+/**
+ * @author Christian Biasuzzi <christian.biasuzzi@techrain.eu>
+ */
+@AutoService(Importer.class)
+public class MatpowerImporter implements Importer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MatpowerImporter.class);
+
+    private static final String FORMAT = "MATPOWER";
+
+    private static final String[] EXTENSIONS = {"m", "mat"};
+
+    public static final LocalDate DEFAULTDATE = LocalDate.of(2020, Month.JANUARY, 1);
+
+    private static final Parameter IGNORE_BASE_VOLTAGE_PARAMETER = new Parameter("ignore-base-voltage",
+            ParameterType.BOOLEAN,
+            "Ignore base voltage specified in the file",
+            Boolean.TRUE);
+
+    private static final class PerUnitContext {
+
+        private final double baseMva; // base apparent power
+
+        private final boolean ignoreBaseMva;
+
+        private PerUnitContext(double baseMva, boolean ignoreBaseMva) {
+            this.baseMva = baseMva;
+            this.ignoreBaseMva = ignoreBaseMva;
+        }
+
+        public boolean isIgnoreBaseMva() {
+            return ignoreBaseMva;
+        }
+
+        private double getBaseMva() {
+            return baseMva;
+        }
+    }
+
+    private static ContainersMapping createContainerMapping(MatpowerModel matpowerModel) {
+        ContainersMapping containersMapping = new ContainersMapping();
+
+        // group buses connected to non impedant lines to voltage levels
+        createVoltageLevelMapping(matpowerModel, containersMapping);
+
+        // group voltage levels connected by transformers to substations
+        createSubstationMapping(matpowerModel, containersMapping);
+
+        return containersMapping;
+    }
+
+    private static boolean isTransformer(MBranch branch) {
+        return branch.getRatio() != 0;
+    }
+
+    private static void createSubstationMapping(MatpowerModel model, ContainersMapping containersMapping) {
+        UndirectedGraph<String, Object> sGraph = new Pseudograph<>(Object.class);
+        for (String voltageLevelId : containersMapping.voltageLevelIdToBusNums.keySet()) {
+            sGraph.addVertex(voltageLevelId);
+        }
+        for (MBranch branch : model.getBranches()) {
+            if (isTransformer(branch)) {
```

Comment:
Question: is there non impedant branches in MatPower format? In that case, could their ends be in the same substation or not?

Reply:
there are not, apparently. fixed